nucypher/network/status/assets/hendrix-update.js

```diff
@@ -0,0 +1,32 @@
+window.onload = function () {
+    const origin_hostname = window.location.hostname;
+    const socket = new WebSocket("ws://"+ origin_hostname +":9000");
```

Comment:
I was hoping to make the port number be configurable by adding another hidden element and getting the value from it eg. for a hidden div `ws_port = document.getElementById('...').innerHTML`. However, there seems to be some timing issue with `window.onload` and when the element is actually rendered - this is also the reason for the null checks below for the hidden buttons.

This script is specified in the footer of the page automatically since the file is in the `assets` folder, so it is weird that when this function is called, the element is not yet loaded - some Dash issue most likely.

Reply:
to just get it to work you could probably do something like
```
window.onload = function () {
const socket;
const origin_hostname = window.location.hostname;
setTimeout(function() {
    socket = new WebSocket("ws://"+ origin_hostname +":9000");
}, 500 <or whatever>);
}
```

Reply:
I get connected to the socket, whose default port is 9000 and that works fine. 

The issue is if anyone ran Moe and used the `--ws_port` option to use a different port, that would be a problem since the `9000` is hardcoded.

Unless you mean setting a timeout for the WebSocket is just good practice anyway...

Reply:
No the setTimeout is definitely _not_ a good practice but it usually hackily solves issues like what you're seeing.

I don't know Dash at all really but couldn't you do something like 
`html.Script("var websocketport={}".format(MoeSocketPort))` in your header function?

Reply:
Also, in my opinion we should make all these hardcoded ports into environment variables.

Reply:
Thanks! I'm definitely not a Dash expert.

There seems to be some issues with the html.Script() component, https://github.com/plotly/dash/issues/93 and it wasn't in their recommendation in https://dash.plot.ly/external-resources, so I haven't really tried to use it. That being said, perhaps I'll give it a whirl.

Reply:
ugh.  good luck!

Reply:
Turns out it was better to just add the script to the page template itself.

Thanks for the discussion!